### ISSUE TITLE
Set permissions on downloaded files/archives

### ISSUE THREAD

**Issue by dadgar** [hashicorp/go-getter#63]
It would be nice to be able to optionally set the permissions on downloaded artifacts.

Cross linking to Nomad issue: https://github.com/hashicorp/nomad/issues/2625

**Comment by jshuping** [comment]
It would be nice to support specifying ownership as well.

**Comment by gauravyadav01** [comment]
Is there any update on this? Are we going to bring it soon? 
thanks

**Comment by mwhooker** [comment]
seems like a nice to have, but should be possible to just run the shell/shell-local provisioner afterwards. I don't think this is something we're interested in supporting at the moment, but happy to review any PRs

**Comment by mwhooker** [comment]
oops, sorry, reopening because I thought this was a packer issue!

**Comment by lgfausak** [comment]
Ran in to this using hashi-nomad, I was expecting the result (unpacking a tar.gz from https) to have the permissions from the tar czpf.  Instead, I get a version of the file that has different file permission bits set, unexpected.

**Comment by rafaelpirolla** [comment]
Bumped into this once again in nomad sadly. :(

**Comment by shantanugadgil** [comment]
Any work going on on this? I keep feeling a need for this for downloaded binaries, for `raw_exec` type of jobs. :eyes: 

Currently I am making do with `chmod +x` in the `command` option of the task.

**Comment by HarryHarcourt** [comment]
> A workaround for permissions is to package the file in a tar or zip archive, permissions are preserved when decompressing.

I don't think this is true, I tried this and it still decompressed it as root.  Would love this to be fixed

**Comment by mister2d** [comment]
Please fix. Please fix. Allow a chmod/chown type abstraction and all will be well to close this issue.

**Comment by danishprakash** [comment]
I'd like to work on this, is there a consensus on whether something like this should be supported or not? cc/ @dadgar

**Comment by shantanugadgil** [comment]
Any work for this? Could this be atleast implemented for the single file download mode? (`mode = file`)

**Comment by danishprakash** [comment]
Totally forgot about this issue. I think I'll take a shot at this over the weekend, hopefully, someone from the team can respond in the meantime.

**Comment by inflatador** [comment]
Any luck on this one? It would make nomad's exec driver a lot more useful.
Thanks everyone!
